interfaces/builtin/cups.go

```diff
@@ -0,0 +1,61 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// On systems where the slot is provided by an app snap, the cups interface is
+// the companion interface to the cups-control interface. The design of these
+// interfaces is based on the idea that the slot implementation (eg cupsd) is
+// expected to query snapd on if the cups-control interface is connected or not
+// and the print service will mediate admin functionality (ie, the rules in
```

Comment:
```suggestion
// expected to query snapd to determine if the cups-control interface is connected or not for the cups client
// and the print service will mediate admin functionality (ie, the rules in
```

Reply:
@anonymouse64 - I went with: `expected to query snapd to determine if the cups-control interface is connected or not for the peer client process and the ...`